Products/ZenUtils/ShardedBTree.py

```diff
@@ -0,0 +1,150 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from BTrees.OOBTree import OOBTree
+from OFS.SimpleItem import SimpleItem
+from persistent.list import PersistentList
+from itertools import chain
+from zlib import adler32
+
+
+DEFAULT_N_SHARDS = 127
+
+
+def default_hash_func(data):
+    """
+    adler32 chosen as hashing mechanism since we dont need a cryptographic hashing
```

Comment:
murmur is not available in the standard library and, after testing it, speed seems to be similar to adler32